tests/components/yeelight/test_config_flow.py

```diff
@@ -0,0 +1,261 @@
+    with patch(f"{MODULE_CONFIG_FLOW}.yeelight.Bulb", return_value=mocked_bulb), patch(
+        f"{MODULE}.async_setup", return_value=True
+    ) as mock_setup, patch(
+        f"{MODULE}.async_setup_entry",
+        return_value=True,
+    ) as mock_setup_entry:
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=config
+        )
+    type(mocked_bulb).get_capabilities.assert_called_once()
+    assert result["type"] == "create_entry"
+    assert result["title"] == DEFAULT_NAME
+    assert result["data"] == {
+        CONF_NAME: DEFAULT_NAME,
+        CONF_IP_ADDRESS: IP_ADDRESS,
+        CONF_TRANSITION: DEFAULT_TRANSITION,
+        CONF_MODE_MUSIC: DEFAULT_MODE_MUSIC,
+        CONF_SAVE_ON_CHANGE: DEFAULT_SAVE_ON_CHANGE,
+        CONF_NIGHTLIGHT_SWITCH: True,
+    }
+    await hass.async_block_till_done()
+    mock_setup.assert_called_once()
+    mock_setup_entry.assert_called_once()
+
+    # Duplicate
+    mocked_bulb = _mocked_bulb()
+    with patch(f"{MODULE_CONFIG_FLOW}.yeelight.Bulb", return_value=mocked_bulb):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": config_entries.SOURCE_IMPORT}, data=config
+        )
+    assert result["type"] == "abort"
+    assert result["reason"] == "already_configured"
+
+
+async def test_manual(hass: HomeAssistant):
+    """Test manually setup."""
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": config_entries.SOURCE_USER}
+    )
+    assert result["type"] == "form"
+    assert result["step_id"] == "user"
+    assert not result["errors"]
+
+    # Cannot connect (timeout)
+    mocked_bulb = _mocked_bulb(cannot_connect=True)
+    with patch(f"{MODULE_CONFIG_FLOW}.yeelight.Bulb", return_value=mocked_bulb):
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_IP_ADDRESS: IP_ADDRESS}
+        )
+    assert result2["type"] == "form"
+    assert result2["step_id"] == "user"
+    assert result2["errors"] == {"base": "cannot_connect"}
+
+    # Cannot connect (error)
+    type(mocked_bulb).get_capabilities = MagicMock(side_effect=OSError)
+    with patch(f"{MODULE_CONFIG_FLOW}.yeelight.Bulb", return_value=mocked_bulb):
+        result3 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_IP_ADDRESS: IP_ADDRESS}
+        )
+    assert result3["errors"] == {"base": "cannot_connect"}
+
+    # Success
+    mocked_bulb = _mocked_bulb()
+    with patch(f"{MODULE_CONFIG_FLOW}.yeelight.Bulb", return_value=mocked_bulb), patch(
+        f"{MODULE}.async_setup", return_value=True
+    ), patch(
+        f"{MODULE}.async_setup_entry",
+        return_value=True,
+    ):
+        result4 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], {CONF_IP_ADDRESS: IP_ADDRESS}
+        )
+    assert result4["type"] == "create_entry"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=6090606)